app/uk/gov/hmrc/exports/models/Pointer.scala

```diff
@@ -0,0 +1,27 @@
+package uk.gov.hmrc.exports.models
+
+import uk.gov.hmrc.exports.models.PointerSectionType.WCOPointerSectionType
+
+case class Pointer(sections: Seq[PointerSection]) {
+  //  Converts a pointer into it's pattern form
+  // e.g. ABC.DEF.*.GHI (if the pointer contains a sequence index)
+  // e.g. ABC.DEF.GHI (if the pointer doesnt contain a sequence)
+  def pattern: String = sections.map(_.pattern).mkString(".")
+
+  // Converts a pointer into its value form
+  // e.g. ABC.DEF.1.GHI  (if the pointer contains a sequence with index 1)
+  // e.g. ABC.DEF.GHI (if the pointer doesnt contain a sequence)
+  def value: String = sections.map(_.value).mkString(".")
+}
+
+case class PointerSection(value: String, `type`: WCOPointerSectionType) {
```

Comment:
This will represent one Pointer section in the XML e.g.
```
<Pointer>
    <DocumentSectionCode>42A</DocumentSectionCode>
</Pointer>
```
is equivalent to `PointerSection("42A", PointerSectionType.FIELD)`

Reply:
For sequences the XML seems to look like
```
<Pointer>
    <SequenceNumeric>1</SequenceNumeric>
    <DocumentSectionCode>68A</DocumentSectionCode>
</Pointer>
```
This would be two `PointerSection`s
`PointerSection("1", PointerSectionType.SEQUENCE)`
`PointerSection("68A", PointerSectionType.FIELD)`

Reply:
For XML nodes with `tagID`s these would be separate `PointerSection`s too
```
<Pointer>
    <DocumentSectionCode>70A</DocumentSectionCode>
    <TagID>166</TagID>
</Pointer>
```
If I understand correctly the `TagID` is just the final section of the error pointer, lets just treat it like any other section
`PointerSection("70A", PointerSectionType.FIELD)`
`PointerSection("166", PointerSectionType.FIELD)`